utils/utils.go

```diff
@@ -586,3 +588,129 @@ func Compress(src string, buf io.Writer) error {
 		return nil
 	})
 }
+
+// Check if a string is purely numeric
+func isNumeric(s string) bool {
+	for _, r := range s {
+		if !unicode.IsDigit(r) {
+			return false
+		}
+	}
+	return true
+}
+
+// Split version into components (numeric and non-numeric) using both '.' and '-'
+func splitVersion(version string) []string {
+	version = strings.ReplaceAll(version, "-", ".")
+	return strings.Split(version, ".")
+}
+
+// Compare two version strings
+func compareVersions(v1, v2 string) int {
+	v1Components := splitVersion(v1)
+	v2Components := splitVersion(v2)
+
+	maxLen := len(v1Components)
+	if len(v2Components) > maxLen {
+		maxLen = len(v2Components)
+	}
+
+	for i := 0; i < maxLen; i++ {
+		var part1, part2 string
+		if i < len(v1Components) {
+			part1 = v1Components[i]
+		}
+		if i < len(v2Components) {
+			part2 = v2Components[i]
+		}
+
+		if isNumeric(part1) && isNumeric(part2) {
+			num1, _ := strconv.Atoi(part1)
+			num2, _ := strconv.Atoi(part2)
+			if num1 != num2 {
+				return num1 - num2
+			}
+		} else if isNumeric(part1) && !isNumeric(part2) {
+			return -1
+		} else if !isNumeric(part1) && isNumeric(part2) {
+			return 1
+		} else {
+			if part1 != part2 {
+				return strings.Compare(part1, part2)
+			}
+		}
+	}
+
+	return 0
+}
+
+// Function to get all version directories sorted in descending order
+func GetAllVersionDirsSortedDesc(modelVersionsDirPath string) ([]string, error) {
```

Comment:
Should we consider to add unit test on this method ?

Reply:
+1. // @Jou